File: backend/app/schemas/pipeline.py
from pydantic import BaseModel


class PipelineStepResponse(BaseModel):
    id: str
    step_type: str
    step_order: int
    status: str
    output_data: str | None = None
    selected_option: str | None = None
    duration_ms: int | None = None
    error_message: str | None = None

    model_config = {"from_attributes": True}


class StepUpdateRequest(BaseModel):
    selected_option: dict | None = None


class PipelineResponse(BaseModel):
    project_id: str
    current_step: int
    steps: list[PipelineStepResponse]

File: frontend/src/types/pipeline.ts
export interface PipelineStep {
  id: string;
  step_type: string;
  step_order: number;
  status: string;
  output_data?: string | null;
  selected_option?: string | null;
  duration_ms?: number | null;
  error_message?: string | null;
}

export interface Pipeline {
  project_id: string;
  current_step: number;
  steps: PipelineStep[];
}
